gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineEclipse.groovy

```diff
@@ -25,11 +26,61 @@ import org.gradle.plugins.ide.eclipse.EclipsePlugin
  */
 class BaselineEclipse extends AbstractBaselinePlugin {
 
+    /**
+     * Copies all name/value pairs from {@code from} to {@code into}, replacing variable names (e.g., $var or ${var})
+     * occurring in values via a {@link SimpleTemplateEngine} and the given binding.
+     */
+    static def mergeProperties(Properties from, Properties into, Map binding) {
+        from.stringPropertyNames().each { property ->
+            def propertyValue = from.getProperty(property)
+            try {
+                // Not all properties are well-formed; attempt to parse and fall-back to original value.
+                // Sadly, this is outrageously slow.
```

Comment:
Didn't measure, but on the order of 2-3 seconds for the 300 lines, vs "immediate" without the template scanning.